### ISSUE TITLE
Hard to parse redis config file due to non section and space separated kv

### ISSUE THREAD

**Issue by naughtyGitCat** [salaros/config-parser#17]
### Want:

parse redis config file to List<KeyValuePair>

### problem
the `config-parser` not friendly to none section and space separated kv config file

### output 
I tried to manually add a section name before the config file begins, and set separator to " ", but it raise exceptions

```c#
                var dict = new Dictionary<string, string>();
                raw.ConfigString = "[Fake]\n" + raw.ConfigString;
                this.logger.LogDebug($"redis config string: \n {raw.ConfigString}");
                var parserConfig = new ConfigParserSettings
                {
                    KeyValueSeparator = " ",
                    MultiLineValues = MultiLineValues.AllowValuelessKeys|MultiLineValues.AllowEmptyTopSection
                };
                var parser = new ConfigParser(raw.ConfigString, parserConfig);
                this.logger.LogDebug($"INI FILE SECTIONS: {string.Join(",", parser.Sections)}");
                this.logger.LogDebug($"INI FILE SECTION KEYS: {string.Join(",", parser.Sections[0].Keys)}");
```

out put:
```bash
 [Fake]
bind 11.1.1.11
port 12345
protected-mode no
daemonize yes
supervised no
loglevel notice
databases 16

tcp-backlog 16000
timeout 0
tcp-keepalive 60
maxclients 16000

maxmemory 6144MB
maxmemory-policy volatile-lru
save 300 10

stop-writes-on-bgsave-error no
rdbcompression yes
rdbchecksum yes

slave-serve-stale-data yes
slave-read-only yes
slave-priority 100
repl-ping-slave-period 10
repl-disable-tcp-nodelay no
min-slaves-to-write 0
min-slaves-max-lag 0

appendonly no
appendfsync everysec
aof-rewrite-incremental-fsync yes
auto-aof-rewrite-percentage 100
auto-aof-rewrite-min-size 64mb
no-appendfsync-on-rewrite no

hash-max-ziplist-entries 512
hash-max-ziplist-value 64

list-max-ziplist-entries 512
list-max-ziplist-value 64

set-max-intset-entries 512

zset-max-ziplist-entries 128
zset-max-ziplist-value 64
activerehashing yes
client-output-buffer-limit normal 0 0 0
client-output-buffer-limit slave 512mb 128mb 120
client-output-buffer-limit pubsub 32mb 8mb 60

cluster-enabled yes
cluster-node-timeout 30000
cluster-slave-validity-factor 10
cluster-migration-barrier 1
cluster-require-full-coverage yes
2020-12-09 10:10:56.487 +08:00 [WRN] =>[XXX.YYY.Redis.Controllers.RedisController]=> FetchInstanceConfig failed, System.ArgumentNullException: Value cannot be null. (Parameter 'separator')
   at Salaros.Configuration.ConfigKeyValue`1..ctor(String keyName, String separator, T value, Int32 lineNumber)
   at Salaros.Configuration.ConfigParser.ReadKeyAndValue(ConfigSection& currentSection, ConfigLine& currentLine, String lineRaw, Int32 lineNumber, Boolean append)
   at Salaros.Configuration.ConfigParser.Read(String configContent)
   at Salaros.Configuration.ConfigParser..ctor(String configFile, ConfigParserSettings settings)
```

**Comment by salaros** [comment]
Hi @naughtyGitCat could you attach here your config file? or is it default Redis config I could have by installing Redis?

**Comment by salaros** [comment]
I will try to use Redis config file you get out of the box (after installing it)
Meanwhile @naughtyGitCat please consider starring and sharing this repo! It helps us getting known and grow the community.

![](https://github.com/salaros/config-parser/raw/master/.github/assets/star_us.gif)
